Select 컴포넌트 리팩토링  및 Informaion 페이지 리팩토링

jma1020 opened the pull request:
# Pull Request
- [x]  Feature
## Summary

Select 컴포넌트 리팩토링 및 오류수정

## Describe your changes
### Select
- 활성화된 Select 제외한  바깥영역 클릭 시 비활성화 기능 추가
- 기존 select tag 제거
- Select 컴포넌트 스타일 수정 Styled-Component 네이밍 수정
- 불필요한 state 제거 및 type 리팩토링


### Information
- information 페이지 Select (성별, 생년월일, 지역) 분리  
- validation error 메세지 추가


### date file
- date 파일 폴더 디렉토리 변경 및 네이밍 리팩토링

### useRegionFetch hooks
- useRegionFetch --> useCityQuery 및 useDistrictQuery 로 변경 및 React Query 로 마이그레이션
- region api 파일 추가

## Screen Shot
![image](https://user-images.githubusercontent.com/45344418/191880897-961c428d-ff10-4223-9c72-1e374a1046a7.png)

validation message
![image](https://user-images.githubusercontent.com/45344418/191880957-b3d0c0c5-f95f-4cf7-a53a-2323bb6c9b5a.png)


## Issue number and link
#38 

## ETC
현재 InformationPage 에서 useCityQuery와 useDistrictQuery 사용으로 인해 selectedCity watch가 남아있어서,  city 부분만 렌더링 최적화가 이루어지지 않았습니다,  추후 괜찮은 구조가 생각나면 해결해보도록 하겠습니다!

jma1020 committed:
⚡️ setValue 속성 추가

jma1020 committed:
⚡️ Custom Select 사용을 위한 설정 추가

jma1020 committed:
💄 임시 class style 추가

jma1020 committed:
💄 Custom Select 스타일 적용

jma1020 committed:
refact: Select 컴포넌트 리팩토링

- 사용하지 않는 options 삭제

jma1020 committed:
feat: Select 커스텀 리팩토링

- 바깥영역 클릭 시 Select close
- 스타일 변경

jma1020 committed:
refact: Label 스타일 리팩토링

jma1020 committed:
fix: 불필요해진 forwardRef 삭제 및 select 태그 관련 code 삭제

jma1020 committed:
fix: 간단한 스타일 수정

seongjin2427 reviewed (approved):
고생하셨습니다! ㅎㅎ

jma1020 committed:
refact: 외부영역 click 을  document 이벤트리스너 방식 --> Background 생성으로 전환

jma1020 committed:
feat: Select 컴포넌트 react hook form 의존성 제거

sangrimlee commented on components/shared/Select/Select.styled.ts:
```diff
@@ -1,11 +1,131 @@
+          pointer-events: all;
+          margin-top: 8px;
+          box-shadow: -1px 1px 2px rgba(67, 70, 74, 0.0001),
+            -2px 2px 5px rgba(67, 86, 100, 0.123689);
```
이런 스타일이 어떻게 적용되는지 PR에 스크린샷 같은 것들로 남겨주어도 좋을 것 같아요!

sangrimlee commented on components/shared/Select/Select.styled.ts:
```diff
@@ -1,11 +1,131 @@
+  width: 100%;
+  color: ${(props) => props.theme.colors.gray[400]};
+  outline: none;
+`;
```
rem이랑 Naming Convention을 지켜주시면 좋을 것 같아요!

jma1020 committed:
feat: z-index 상수파일 생성

jma1020 committed:
feat:Select 컴포넌트  화살표 SVG 파일 추가

jma1020 committed:
feat: onCahngeOption 함수 생성

- setValue 공용으로 사용하기위해 생성

jma1020 committed:
feat: Select 컴포넌트 수정

- Styled 컴포넌트 네이밍 변경
- onChangeOption props 수정
- 리팩토링

jma1020 committed:
feat: styled 수정

jma1020 committed:
feat: check Icon Svg 파일 추가

jma1020 committed:
refact: Select 리팩토링

- style 수정
- 불필요한 props 삭제
- disabled style 추가
- selected style 추가
- 범주에서 벗어난 경우 null로 변경

jma1020 committed:
refact: Select 컴포넌트 제네릭 타입 추가

- 제네릭 타입 추가

jma1020 committed:
Merge branch 'develop' into feature/Select-Custom

jma1020 committed:
feat: 지역 데이터 React Query 적용

- 기존 useRegionFetch 삭제
- useCity useDistrict 세분화 및 React Query 적용

jma1020 committed:
refact: 파일 네이밍 변경

jma1020 committed:
refact: 파일 네이밍 변경

jma1020 committed:
refact: api 코드 및 파일 분리

jma1020 committed:
refact: Select 및 information 페이지 리팩토링

- Select 컴포넌트 selectedValue 상태 제거
- information에서 watch로 상태 Select 컴포넌트로 전달
- validation 코드 수정
- date 함수 수정

jma1020 committed:
refact: constants 폴더에 date 함수 --> utils 폴더로 이동

jma1020 committed:
refact: information 페이지를  auth 폴더 하위로 이동

- /information  --> /auth/information

jma1020 committed:
refact: utils / date 파일 리팩토링

- 네이밍 변경
- 범용성을 위해 OptionYear range 설정추가

jma1020 committed:
refact: information 리팩토링

- 성별, 생년월일, 지역 Select를 분리
- FormProvider 적용

sangrimlee reviewed (approved):
고생하셨어요 👍🏼

jma1020 committed:
refact: utils 파일 type 분리

seongjin2427 reviewed (approved):
고생하셨습니다!!

jma1020 committed:
fix : types 폴더 생성에 따른 import 수정

jma1020 committed:
Merge branch 'develop' into feature/Select-Custom
